clients/src/main/java/org/apache/kafka/common/record/OffsetAndTimestamp.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.record;
+
+import java.util.Objects;
+
+/**
+ * A container class for offset and timestamp.
+ */
+public final class OffsetAndTimestamp {
+    private final long timestamp;
```

Comment:
The case I was concerned about was: https://github.com/apache/kafka/pull/1852/files#diff-d45970e44e2636ec847b63ac71827b71R641

Are you saying that we cannot hit that case in the `offsetsForTimes` case (only in `endOffsets`)?